As there are existing submitted transactions against item Test TDS Item, you can not change the value of Has Serial No.

### Information about bug

Stock Module - Material Request:
Testcases failed :-
test_mr_to_po_pi_with_serial_nos_TC_B_158
test_mr_to_po_pr_with_multiple_serial_nos_TC_B_157
test_mr_to_po_pr_with_serial_no_TC_B_156

### Module

stock

### Version

Frappe Framework: v15.51.2
ERPNext: v15.50.

### Installation method

manual install

### Relevant log output / Stack trace / Full Error Message.

```shell
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/stock/doctype/material_request/test_material_request.py", line 5951, in test_mr_to_po_pr_with_serial_no_TC_B_156
    item.save()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 342, in save
    return self._save(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 378, in _save
    self.run_before_save_methods()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 1103, in run_before_save_methods
    self.run_method("validate")
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 974, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 1334, in composer
    return composed(self, method, *args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 1316, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 971, in fn
    return method_object(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/stock/doctype/item/item.py", line 208, in validate
    self.cant_change()
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/stock/doctype/item/item.py", line 971, in cant_change
    frappe.throw(msg, title=_("Linked with submitted documents"))
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 603, in throw
    msgprint(
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 568, in msgprint
    _raise_exception()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 519, in _raise_exception
    raise exc
frappe.exceptions.ValidationError: As there are existing submitted transactions against item Test TDS Item, you can not change the value of Has Serial No.Example of a linked document: Purchase Receipt _Test Supplier
```

@Barathpalanisamy 
still failing 
test_mr_to_po_pr_with_serial_no_TC_B_156
test_mr_to_po_pr_with_multiple_serial_nos_TC_B_157

```
FAIL: test_mr_to_po_pr_with_serial_no_TC_B_156 (erpnext.stock.doctype.material_request.test_material_request.TestMaterialRequest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/stock/doctype/material_request/test_material_request.py", line 6342, in test_mr_to_po_pr_with_serial_no_TC_B_156
    self.assertEqual(sle[0]["valuation_rate"], 1000)
AssertionError: 62900.0 != 1000

```

--- a/erpnext/accounts/doctype/pos_invoice/test_pos_invoice.py
+++ b/erpnext/accounts/doctype/pos_invoice/test_pos_invoice.py
@@ -1022,9 +1022,11 @@ class TestPOSInvoice(unittest.TestCase):
 		test_user, pos_profile = init_user_and_profile()
 		opening_entry = create_opening_entry(pos_profile=pos_profile, user=test_user.name)
 		self.assertEqual(opening_entry.status, "Open")
-		inv_points = create_pos_invoice(rate=10000, do_not_save=1)
-
 		frappe.db.set_value("Customer","_Test Customer",'loyalty_program','Test Single Loyalty')
+		inv_points = create_pos_invoice(rate=10000, do_not_save=1)
+		inv_points.save()
+		inv_points.paid_amount = inv_points.grand_total
+		inv_points.submit()
 		before_lp_details = get_loyalty_program_details_with_points(
 			"_Test Customer", loyalty_program="Test Single Loyalty"
 		)
@@ -1162,9 +1164,11 @@ class TestPOSInvoice(unittest.TestCase):
 					"collection_rules": [{"tier_name": "Silver", "collection_factor": 1000, "min_spent": 1000}],
 				}
 			).insert()
+		frappe.db.set_value("Customer","_Test Customer",'loyalty_program','Test Single Loyalty')
 		inv_points = create_pos_invoice(rate=10000, do_not_save=1)
 		inv_points.save()
-		frappe.db.set_value("Customer","_Test Customer",'loyalty_program','Test Single Loyalty')
+		inv_points.paid_amount = inv_points.grand_total
+		inv_points.submit()
 		before_lp_details = get_loyalty_program_details_with_points(
 			"_Test Customer", loyalty_program="Test Single Loyalty"
 		)

--- a/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
+++ b/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
@@ -7226,6 +7226,7 @@ def create_company_and_supplier():
 				"company_name": parent_company,
 				"abbr": "TC-1",
 				"default_currency": "INR",
+				"country":"India",
 				"is_group": 1,
 				"gstin": "27AAAAP0267H2ZN",
 				"gst_category": "Registered Regular"
@@ -7243,6 +7244,7 @@ def create_company_and_supplier():
 				"company_name": child_company,
 				"abbr": "TC-3",
 				"default_currency": "INR",
+				"country":"India",
 				"gstin": "27AABCT1296R1ZN",
 				"gst_category": "Registered Regular",
 				"parent_company": parent_company

--- a/erpnext/manufacturing/doctype/blanket_order/test_blanket_order.py
+++ b/erpnext/manufacturing/doctype/blanket_order/test_blanket_order.py
@@ -268,6 +268,7 @@ class TestBlanketOrder(FrappeTestCase):
 			purpose="Material Receipt"
 		)
 		so.delivery_date = add_days(nowdate(), 5)
+		so.currency = "INR"
 		so.submit()
 
 		bo.reload()

--- a/erpnext/selling/doctype/sales_order/test_sales_order.py
+++ b/erpnext/selling/doctype/sales_order/test_sales_order.py
@@ -5992,7 +5992,7 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		customer = create_customer("_Test Customer 1",currency = "INR")
 		make_item("_Test Item", {"is_stock_item": 1})
 		make_stock_entry(item_code="_Test Item", qty=100, rate=500, target="_Test Warehouse - _TC")
-		sales_order = make_sales_order(customer=customer,cost_center='Main - _TC', selling_price_list='Standard Selling', do_not_save=True)
+		sales_order = make_sales_order(customer=customer,cost_center='Main - _TC', selling_price_list='_Test Price List', do_not_save=True)
 		sales_order.delivery_date = nowdate()
 		if qty and rate:
 			for item in sales_order.items:
@@ -6031,7 +6031,7 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			customer="_Test Registered Customer",
 			warehouse="Stores - _TIRC",
 			cost_center="Main - _TIRC",
-			selling_price_list="Standard Selling",
+			selling_price_list='_Test Price List',
 			item_code=item_code,
 			qty=qty,
 			rate=rate,

--- a/erpnext/selling/page/point_of_sale/point_of_sale.py
+++ b/erpnext/selling/page/point_of_sale/point_of_sale.py
@@ -207,7 +207,7 @@ def get_items(start, page_length, price_list, item_group, pos_profile, search_te
 				"selling": True,
 				"valid_from": ["<=", current_date],
 			},
-			 or_filters=[
+			or_filters=[
 			["valid_upto", "is", "not set"], 
 			["valid_upto", "=", ""],
 			["valid_upto", ">=", current_date]

--- a/erpnext/stock/doctype/material_request/test_material_request.py
+++ b/erpnext/stock/doctype/material_request/test_material_request.py
@@ -6319,8 +6319,10 @@ class TestMaterialRequest(FrappeTestCase):
 
 		po = make_purchase_order(mr.name)
 		po.supplier = supplier
-		po.items[0].rate = 1000
 		po.insert()
+		po.currency = "INR"
+		po.items[0].rate = 1000
+		po.save()
 		po.submit()
 
 		pr = make_purchase_receipt(po.name)
@@ -6380,8 +6382,10 @@ class TestMaterialRequest(FrappeTestCase):
 
 		po = make_purchase_order(mr.name)
 		po.supplier= supplier
-		po.items[0].rate = 1000
 		po.insert()
+		po.currency = "INR"
+		po.items[0].rate = 1000
+		po.save()
 		po.submit()
 
 		pr1 = make_purchase_receipt(po.name)

